tests/components/switch_as_x/test_light.py

```diff
@@ -0,0 +1,174 @@
+    assert state.attributes.get("white_value") is None
+    assert state.attributes.get("effect_list") is None
+    assert state.attributes.get("effect") is None
+    assert state.attributes.get(ATTR_SUPPORTED_COLOR_MODES) == [COLOR_MODE_ONOFF]
+    assert state.attributes.get(ATTR_COLOR_MODE) is None
+
+
+async def test_light_service_calls(hass):
+    """Test service calls to light."""
+    await async_setup_component(hass, "switch", {"switch": [{"platform": "demo"}]})
+    config_entry = MockConfigEntry(
+        data={},
+        domain=DOMAIN,
+        options={"entity_id": "switch.decorative_lights", "target_domain": "light"},
+        title="decorative_lights",
+    )
+    config_entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert hass.states.get("light.decorative_lights").state == "on"
+
+    await common.async_toggle(hass, "light.decorative_lights")
+
+    assert hass.states.get("switch.decorative_lights").state == "off"
+    assert hass.states.get("light.decorative_lights").state == "off"
+
+    await common.async_turn_on(hass, "light.decorative_lights")
+
+    assert hass.states.get("switch.decorative_lights").state == "on"
+    assert hass.states.get("light.decorative_lights").state == "on"
+    assert (
+        hass.states.get("light.decorative_lights").attributes.get(ATTR_COLOR_MODE)
+        == COLOR_MODE_ONOFF
+    )
+
+    await common.async_turn_off(hass, "light.decorative_lights")
+    await hass.async_block_till_done()
+
+    assert hass.states.get("switch.decorative_lights").state == "off"
+    assert hass.states.get("light.decorative_lights").state == "off"
+
+
+async def test_switch_service_calls(hass):
+    """Test service calls to switch."""
+    await async_setup_component(hass, "switch", {"switch": [{"platform": "demo"}]})
+    config_entry = MockConfigEntry(
+        data={},
+        domain=DOMAIN,
+        options={"entity_id": "switch.decorative_lights", "target_domain": "light"},
+        title="decorative_lights",
+    )
+    config_entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert hass.states.get("light.decorative_lights").state == "on"
+
+    await switch_common.async_turn_off(hass, "switch.decorative_lights")
+    await hass.async_block_till_done()
+
+    assert hass.states.get("switch.decorative_lights").state == "off"
+    assert hass.states.get("light.decorative_lights").state == "off"
+
+    await switch_common.async_turn_on(hass, "switch.decorative_lights")
+    await hass.async_block_till_done()
+
+    assert hass.states.get("switch.decorative_lights").state == "on"
+    assert hass.states.get("light.decorative_lights").state == "on"
+
+
+@pytest.mark.parametrize("target_domain", ("light",))
+async def test_config_entry(hass: HomeAssistant, target_domain):
```

Comment:
This name seems to generic. All the platform tests use a config entry.

Reply:
Improved in https://github.com/home-assistant/core/pull/67951